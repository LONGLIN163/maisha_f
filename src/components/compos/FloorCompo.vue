<template>
    <div>
        <div class="floor">
            <div class="floor-title">{{floorTitle}}</div>
            <div class="floor-anomaly">
                <van-row>
                    <van-col span="12" class="floor-one">
                        <img :src="floorData0.image" alt="">
                    </van-col>
                    <van-col span="12" class="floor-two">
                        <div>
                            <div><img :src="floorData1.image"/></div>
                            <div><img :src="floorData2.image"/></div>
                        </div>
                    </van-col>
                </van-row>
            </div>

            <div class="floor-rule">
                <div v-for="(item ,index) in floorData.slice(3)" :key="index">
                    <img :src="item.image" height="100%" width="100%"/>
                </div>
            </div>
 
        </div>

    </div>
</template>

<script>
    export default {
        props:['floorData','floorTitle'],
        data() {
            return {
                floorData0:{},
                floorData1:{},
                floorData2:{},
            }
        },
        created(){
        },
        mounted(){
        },
        watch:{ 
            //monitor asyn data
            floorData:function() { 
                this.floorData0=this.floorData[0]
                this.floorData1=this.floorData[1]
                this.floorData2=this.floorData[2]
            }
        }
        
    }
</script>

<style lang="scss" scoped>

  .floor-title{
    text-align: center;
    font-size: 14px;
    line-height: 1.8rem;
    height: 1.8rem;
    padding-top: 10px;
  }

  .floor-anomaly{
    height: 10rem;
    .floor-one {
        border-right:1px solid #ddd;
        height: 10rem;
        img {
             width: 100%;
             height:100%;
        }
    }
    .floor-two{
        height: 10rem;
        div{
            height: 5rem;
            img{
                width: 100%;
                height: 100%;
            }
        }
    }
  }

  .floor-rule{
    display: flex;
    flex-direction: row;
    flex-wrap:wrap;
    background-color: #fff;
    height: 8rem;
    div{
        -webkit-box-sizing: border-box;
        box-sizing: border-box;
        width:50%;
        border-bottom:1px solid #ddd;
      }
    div:nth-child(odd){
        border-right: 1px solid #ddd;
    }
  }

</style>