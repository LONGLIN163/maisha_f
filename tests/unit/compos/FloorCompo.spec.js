import {mount, createLocalVue} from '@vue/test-utils'
import FloorCompo from '@/components/compos/FloorCompo.vue'
import {fakeFloordata} from '../../mockData'

const Vant=require("vant")
const localVue = createLocalVue();
localVue.use(Vant); 

describe('FloorCompo', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(FloorCompo,{
      localVue,
      propsData: {
        floorData:fakeFloordata,
        floorTitle:"Snack"
      }
    })
  })

  it('should renders FloorCompo', () => {
    expect(wrapper).toMatchSnapshot()
  })
})