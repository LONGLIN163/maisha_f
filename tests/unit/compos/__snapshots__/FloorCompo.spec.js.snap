// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorCompo should renders FloorCompo 1`] = `
<div>
  <div class="floor">
    <div class="floor-title">Snack</div>
    <div class="floor-anomaly">
      <div class="van-row">
        <div class="floor-one van-col van-col--12"><img alt=""></div>
        <div class="floor-two van-col van-col--12">
          <div>
            <div><img></div>
            <div><img></div>
          </div>
        </div>
      </div>
    </div>
    <div class="floor-rule"></div>
  </div>
</div>
`;
